app/models/user.rb

```diff
@@ -121,7 +121,7 @@ def moderatable
   # after_validation :reset_lost_password_token
 
   after_save :nullify_authentication_id, if: :any_sso_authorizations?
-  after_commit :notify_deletion, on: :destroy
+  after_destroy :archive_as_deleted
```

Comment:
Because the account must still be present :smile: and this should be part of the transaction.

Reply:
couldn't be audit used for this?

Reply:
I am not sure if that is a good idea because it has a very different purpose :thinking: It could be an option but should be very well thought.

Reply:
I feel that is very scary because:
1. We might remove the audit before time and not be able to use it for this at some point.
2. Making queries in the audit is not really ideal.
3. We use the audits as a log mostly, not to "use to do X thing, and then remove once used", which would change a lot.